src/librbd/internal.cc

```diff
@@ -413,6 +413,58 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
     rados_completion->release();
   }
 
+  int rollback_parent(ImageCtx *ictx, uint64_t snap_id)
+  {
+    assert(ictx);
+    assert(ictx->parent_lock.is_locked());
+    assert(ictx->snap_lock.is_locked());
+
+    CephContext *cct = ictx->cct;
+    int r = 0;
+    std::map<librados::snap_t, SnapInfo>::const_iterator it = ictx->snap_info.find(snap_id);
+    assert(it != ictx->snap_info.end());
+    const SnapInfo& snap_info(it->second);
+    if (snap_info.parent.spec.snap_id == CEPH_NOSNAP) {
```

Comment:
Minor: while technically correct to check ```snap_id == CEPH_NOSNAP```, since the rest of librbd uses ```pool_id == -1``` for the same check, it might make it clearer (here and below)